### PR TITLE
Add clean-changelog-unreleased step to release script

### PR DESCRIPTION
**Pull request by sosukesuzuki**
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/release/steps/clean-changelog-unreleased.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  ---
+  author: "🚧"
+  authorURL: "https://github.com/🚧"
+  title: "Prettier 🚧"
+  ---
+
+  🚧 Write an introduction here.
+
+`
+  );
```
Comment (fisker) [inline scripts/release/steps/clean-changelog-unreleased.js:29]: On draft-blog-post step, copy `blog-post-intro.md` to `blog-post-intro-2.2.0.md`, use if, and in this step remove `blog-post-intro-2.2.0.md`

Comment (fisker) [reply scripts/release/steps/clean-changelog-unreleased.js:29]: Oh, in that step, there is no version info, so `blog-post-intro-releasing.md` or something.

File: scripts/release/steps/clean-changelog-unreleased.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      if (!categories.has(dir.name)) {
+        throw new Error(`Unknown category: ${dir.name}`);
+      }
+      fs.readdirSync(dirPath)
+        .filter((fileName) => /^pr-\d+\.md$/.test(fileName))
+        .map((fileName) => path.join(dirPath, fileName))
+        .forEach((file) => {
+          fs.unlinkSync(file);
+        });
+    });
```
Comment (fisker) [inline scripts/release/steps/clean-changelog-unreleased.js:65]: globby('*/*.md', {cwd: changelogUnreleasedDir})

File: scripts/release/steps/clean-changelog-unreleased.js (hunk 1 above)
Comment (sosukesuzuki) [reply scripts/release/steps/clean-changelog-unreleased.js:30]: Do you mean to create a draft blog post in the release script?

Comment (fisker) [reply scripts/release/steps/clean-changelog-unreleased.js:30]: Update "scripts/draft-blog-post.js"

```js
const introFile = path.join(changelogUnreleasedDir, "blog-post-intro-releasing.md");
if (!fs.exitsSync(introFile)) {
  // Copy "blog-post-intro.md" to "blog-post-intro-releasing.md"
}
```

Here, just delete `changelog_unreleased/blog-post-intro-releasing.md` file.

Meaning we don't use "blog-post-intro.md" file directly, it's used as a template

Comment (sosukesuzuki) [reply scripts/release/steps/clean-changelog-unreleased.js]: I've added two changes:
1. Add `BLOG_POST_INTRO_TEMPLATE.md` and make` blog-post-intro.md` a file that can be edited directly.
2. Remove `clean-changelog-unreleased` from the release script to make it a standalone script.

File: scripts/release/package.json [hunk 3]
```diff
@@ -3,6 +3,7 @@
   "dependencies": {
     "chalk": "2.4.1",
     "execa": "0.10.0",
+    "globby": "11.0.1",
```
Comment (sosukesuzuki) [inline scripts/release/package.json:6]: Oh, I'll remove this

Comment (sosukesuzuki) [reply scripts/release/package.json]: 732f366

File: scripts/clean-changelog-unreleased.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const globby = require("globby");
+
+const changelogUnreleasedDir = path.join(__dirname, "../changelog_unreleased");
+
+// Remove blog-post-intro.md
+const introFile = path.join(changelogUnreleasedDir, "blog-post-intro.md");
+fs.unlinkSync(introFile);
+
+// Remove change files
+const files = globby.sync("*/*.md", { cwd: changelogUnreleasedDir });
```
Comment (fisker) [inline scripts/clean-changelog-unreleased.js:16]:
```suggestion
const files = globby.sync(["blog-post-intro.md", "*/*.md"], {
  cwd: changelogUnreleasedDir,
});
```

Remove `blog-post-intro.md` too?

Comment (fisker) [reply scripts/clean-changelog-unreleased.js:16]: And `globby` support `absolute` options, no need join yourself.

File: scripts/clean-changelog-unreleased.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+const fs = require("fs");
+const path = require("path");
+const globby = require("globby");
+
+const changelogUnreleasedDir = path.join(__dirname, "../changelog_unreleased");
+
+// Remove blog-post-intro.md
+const introFile = path.join(changelogUnreleasedDir, "blog-post-intro.md");
+fs.unlinkSync(introFile);
```
Comment (fisker) [inline scripts/clean-changelog-unreleased.js:13]: Sorry, didn't see this line, just ignore another comment.

**Review by fisker — APPROVED** [review]

**Pull request merged**